terminus/geometry/circle.py

```diff
@@ -45,45 +45,66 @@ def __eq__(self, other):
     def __hash__(self):
         return hash((self.center, self.radius))
 
+    def radius(self):
+        return self.radius
+
+    def center(self):
+        return self.center
+
     def almost_equal_to(self, other, decimals=5):
         return self.center.almost_equal_to(other.center, decimals) and abs(self.radius - other.radius) < 1e-5
 
-    def intersection(self, other):
-        d = (other.center - self.center).norm()
-        r1 = self.radius
-        r2 = other.radius
-        if d == 0 and r1 == r2:
+    def intersection(self, other, buffer=1e-7, decimals=7):
+        p = self.center
+        q = other.center
+        A = self.radius
+        B = other.radius
```

Comment:
I know and getting good names is hard indeed (https://martinfowler.com/bliki/TwoHardThings.html). In this case not only you have to get meaningful names, but also try to keep things short. Temp variables are your friends also, but you should not abuse them. Check also my other comment on avoiding coding many times the seme things.